ts-client/src/dlmm/helpers/index.ts

```diff
@@ -197,8 +199,69 @@ export async function chunkedGetMultipleAccountInfos(
   return accountInfos;
 }
 
-export const computeBudgetIx = () => {
-  return ComputeBudgetProgram.setComputeUnitLimit({
-    units: 1_400_000,
+/**
+ * Gets the estimated compute unit usage with a buffer.
+ * @param connection A Solana connection object.
+ * @param instructions The instructions of the transaction to simulate.
+ * @param feePayer The public key of the fee payer.
+ * @param buffer The buffer to add to the estimated compute unit usage. Max value is 1. Default value is 0.1 if not provided, and will be capped between 50k - 200k.
+ * @returns The estimated compute unit usage with the buffer.
+ */
+export const getEstimatedComputeUnitUsageWithBuffer = async (
+  connection: Connection,
+  instructions: TransactionInstruction[],
+  feePayer: PublicKey,
+  buffer?: number
+) => {
+  if (!buffer) {
+    buffer = 0.1;
+  }
+  // Avoid negative value
+  buffer = Math.max(0, buffer);
+  // Limit buffer to 1
+  buffer = Math.min(1, buffer);
+
+  const estimatedComputeUnitUsage = await getSimulationComputeUnits(
+    connection,
+    instructions,
+    feePayer,
+    []
+  );
+
+  let extraComputeUnitBuffer = estimatedComputeUnitUsage * buffer;
+  if (extraComputeUnitBuffer > MAX_CU_BUFFER) {
```

Comment:
this util should not limit to `add liquidity` function?

Reply:
It applies to all functions. It clamps only the extra CU buffer, not the estimated CU used.